website/docs/tutorials/07-testing/04-e2e.md

```diff
@@ -0,0 +1,62 @@
+# End-to-end tests
+
+There are three main types of testing software: unit tests, integration tests and end-to-end tests.
+
+Protostar testing framework provides an environment to write unit tests for Cairo smart contracts
+in Cairo language itself.
+With help of cheatcodes like [`deploy_contract`] and [`invoke`] it is also easy to write
+integration tests, deploying single or a few contracts and examining internal behaviour of such
+small systems.
+
+## Recommended approach
+
+In end-to-end testing, the goal is to examine interactions not only within a system of contracts
+deployed on the blockchain, but also to examine interactions triggered by external services,
+like web frontends, and observing effects happening within these services themselves.
+
+:::note
+As an example, an end-to-end test of a voting application could trigger a click on a _Vote_ button,
+causing a transaction being performed on the blockchain.
+The test then waits for the transaction to be accepted, and examines changes of observable state of
+voting contract, and the results gathered by the voting application.
+:::
+
+It's easy to observe, that end-to-end testing is more focused on controlling applications
+interacting with smart contracts, while the manipulation of latter is limited only to deploying,
+presetting a given state and then observing state changes.
+Therefore, end-to-end tests suit more test suites of client applications, implemented in their
+technologies, rather than Protostar-powered smart contracts' test suites.
+The use cases needed for setting up smart contracts in a test can be fulfilled with three tools:
```

Comment:
```suggestion
End-to-end testing should focus on the bigger picture of the whole system not only the smart contracts. 
Modifications and readings of the on-chain state should be done thourgh exposed interfaces of contracts.
Therefore, we recommed to include end-to-end tests in test suites of your client applications, implemented in their technologies. We suggest to use following tools to set up described testing environment:
```

Reply:
I don't like saying `we` in documentation, so I will edit this further to use passive voice.